lib/integration/framework/Test/Integration/Framework/Setup.hs

```diff
@@ -342,67 +338,67 @@ recordPoolGarbageCollectionEvents TestingCtx{..} eventsRef =
 
 withServer
     :: TestingCtx
-    -> Config
     -> FaucetFunds
     -> Pool.DBDecorator IO
-    -> ( T.Text
-         -> CardanoNodeConn
-         -> NetworkParameters
-         -> URI
-         -> IO ()
-       )
-    -> IO ExitCode
+    -> ContT () IO (T.Text, NetworkParameters, URI, RunFaucetQ IO)
 withServer
     ctx@TestingCtx{..}
-    clusterConfig
     faucetFunds
-    dbDecorator
-    onReady =
-        bracketTracer' tr "withServer" $ do
-            let tr' = cfgTracer clusterConfig
-            withSMASH tr' (toFilePath . absDirOf $ testDir) $ \smashUrl -> do
-                withCluster clusterConfig faucetFunds
-                    $ onClusterStart
-                        ctx
-                        (onReady (T.pack smashUrl))
-                        dbDecorator
+    dbDecorator = do
+        _ <- ContT $ \k -> bracketTracer' tr "withServer" $ k ()
+        ((runMonitorQ, runFaucetQ), ToTextTracer clog) <-
+            withLocalCluster "main-integration-tests" NoWallet
+                defaultEnvVars faucetFunds
+        smashUrl <- ContT $ withSMASH clog (toFilePath . absDirOf $ testDir)
+        (np, uri) <-
+            onClusterStart
+                ctx
+                dbDecorator
+                runMonitorQ
```

Comment:
I'm wondering — do we want the `--wallet-present` option for the `local-cluster` executable at all? 🤔

I think that it's fair game to have the faucet be part of the `local-cluster`, but I don't see a good conceptual reason why the `local-cluster` needs to start a `cardano-wallet` process as well.

As far as the test setup code here is concerned, we cann `local-cluster` with `NoWallet` anyway, and instead start up a wallet in `onClusterStart`.

Reply:
Ticket ?

Reply:
> The only good reason why we want it now is that wallet-e2e are using it
> I am open to fix / remove wallet-e2e package

Ah, I see. Hm, the `wallet-e2e` package may contain interesting ideas, but it currently doesn't test anything interesting. 🤔 As a minimum viable refactoring, I would keep the package around, but change our CI to no longer run the tests from the `wallet-e2e` package, allowing us to remove the command line option `--wallet-present`.

> Ticket ?

Feel free to create a ticket — we can discuss tomorrow if needed.